cmd/kube-apiserver/app/encryption/encryption.go

```diff
@@ -100,17 +123,18 @@ func ParseEncryptionConfiguration(f io.Reader) (map[schema.GroupResource]value.T
 	return result, nil
 }
 
-// GetPrefixTransformer constructs and returns the appropriate prefix transformers for the passed resource using its configuration
-func GetPrefixTransformers(config *ResourceConfig) ([]value.PrefixTransformer, error) {
```

Comment:
> Are there real uses of this outside of kube-apiserver? I think the generic layer should not  mandate a particular config file format. The configuration structs and constructors are OK. The issue here is that @sakshamsharma wants to add a callout which I don't want the generic library to add a dependency for. (Users of the apiserver library shouldn't have to take a dependency on gce client.)

Providing a general format for configuration of its flags seems fully in the purview of the genericapiserver package.  Nothing forces people to use it, but it is a strong suggestion.  Similar to the manner in which admission config is specified.

A genericapiserver can choose to do something different, but life will be much easier if the "normal" interactions are consistent.